### ISSUE TITLE
forChild executing before forRoot on Custom Module (Repo available inside)

### ISSUE THREAD

**Issue by appsolutegeek** [nestjs/nest#1827]
[ ] Regression 
[x] Bug report
[ ] Feature request
[x] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
I am having a problem with a race condition. I have created a ConfigModule - this has a forRoot and a forChild. 

The forRoot sets up the loading of the .env file and the forChild uses it in another module.

The problem is that forChild is called before forRoot. The ConfigService would be injected with missing config because forRoot hasn't executed first.

AppModule > ConfigModule.forRoot
InstanceModule > ConfigModule.forChild

I placed some simple console.log commands that output this

```
I am in Config Module  FOR CHILD
I am in Config Module  FOR ROOT
```

As you can see the forChild is being executed first, I tried using forwardRef and that didn't work.

If you let the application run you will see

```
[2019-03-24T11:49:33.602] [ERROR] ConfigService - There are missing mandatory configuration: Missing PORT
[2019-03-24T11:49:33.602] [FATAL] ConfigService - Missing mandatory configuration, cannot continue!, exiting
``` 

This is because I check that some process.env are available which are loaded in via dotenv. Of course, because the forRoot isn't executed first then the forChild returns its own new instance of the ConfigService.

ConfigService validates the availability of the environment variables.

So, basically, the forChild is executing and returning its own ConfigService before forRoot.

TO make it work, if you comment out the InstanceModule inside the AppModule then it will automatically start listening and returns the port number from an environment variable.

Of course, because InstanceModule uses the forChild - there is a race condition.

## Expected behavior

forChild should not be executed before forRoot has finished.

## Minimal reproduction of the problem with instructions

Here is a small repo
https://github.com/appsolutegeek/race-condition-nest

Places to look are AppModule, ConfigModule and InstanceModule

use to execute 

npm run start

## Environment

<pre><code>
Nest version: 6
 
For Tooling issues:
- Node version: 11
- Platform:  Mac

</code></pre>

**Comment by kamilmysliwiec** [comment]
Please, use [StackOverflow](https://stackoverflow.com/questions/tagged/nestjs) for such questions. This is not a bug, but rather a misunderstanding of the framework' module resolution system.
